dav-controller/api/core/acapy/client.py

```diff
@@ -0,0 +1,215 @@
+        schema_version: str = None,
+    ):
+        result = {
+            "name": name,
+            "nonce": nonce,
+            "version": "1.0",
+            "requested_attributes": {},
+            "requested_predicates": {},
+        }
+        d = datetime.today()
+        birth_date = datetime(d.year - age, d.month, d.day)
+        birth_date_format = "%Y%m%d"
+        req_attrib_ident = f"0_{attrib_ident}_uuid"
+        req_predicate_ident = f"0_{predicate_ident}_uuid"
+        if schema_name and schema_version:
+            result["requested_attributes"][req_attrib_ident] = {
+                "names": attrib_names_list,
+                "restrictions": [
+                    {
+                        "schema_name": schema_name,
+                        "schema_version": schema_version,
+                    }
+                ],
+            }
+            result["requested_predicates"][req_predicate_ident] = {
+                "name": "birthdate_dateint",
+                "p_type": ">",
+                "p_value": int(birth_date.strftime(birth_date_format)),
+                "restrictions": [
+                    {
+                        "schema_name": schema_name,
+                        "schema_version": schema_version,
+                    }
+                ],
+            }
+        else:
+            result["requested_attributes"][req_attrib_ident] = {
+                "names": attrib_names_list,
+                "restrictions": [
+                    {
+                        "cred_def_id": cred_def_id,
+                    }
+                ],
+            }
+            result["requested_predicates"][req_predicate_ident] = {
+                "name": "birthdate_dateint",
+                "p_type": ">",
+                "p_value": int(birth_date.strftime(birth_date_format)),
+                "restrictions": [
+                    {
+                        "cred_def_id": cred_def_id,
+                    }
+                ],
+            }
+        if revocation:
+            result["requested_attributes"][req_attrib_ident]["non_revoked"] = {
+                "from": int(time.time()),
+                "to": int(time.time()),
+            }
+            result["requested_predicates"][req_predicate_ident]["non_revoked"] = {
+                "from": int(time.time()),
+                "to": int(time.time()),
+            }
+        return result
+
+    def create_presentation_request(
+        self, presentation_request_configuration: dict = None
+    ) -> CreatePresentationResponse:
+        logger.debug(">>> create_presentation_request")
+        if presentation_request_configuration:
+            present_proof_payload = {
+                "proof_request": presentation_request_configuration
+            }
+        else:
+            present_proof_payload = {
+                "proof_request": self.generate_verification_proof_request(
+                    schema_name="Person", schema_version="1.0"
```

Comment:
This is the only spot I can find a definition for the proof-request configuration to be used. It needs to be configurable by configuration file, not hard-coded otherwise it won't be possible to update/tweak it when necessary.